cmd/performance-profile-creator/cmd/root.go

```diff
@@ -48,28 +56,31 @@ var rootCmd = &cobra.Command{
 		if err != nil {
 			return fmt.Errorf("Error obtaining Nodes %s: %v", mcpName, err)
 		}
-
+		splitReservedCPUsAcrossNUMA, err := strconv.ParseBool(cmd.Flag("split-reserved-cpus-across-numa").Value.String())
+		if err != nil {
+			return fmt.Errorf("Error parsing split-reserved-cpus-across-numa flag: %v", err)
+		}
+		reservedCPUCount, err := strconv.Atoi(cmd.Flag("reserved-cpu-count").Value.String())
+		if err != nil {
+			return fmt.Errorf("Error parsing reserved-cpu-count flag: %v", err)
+		}
 		matchedNodes, err := profilecreator.GetMatchedNodes(nodes, labelSelector)
 		for _, node := range matchedNodes {
 			nodeName := node.GetName()
 			log.Infof("%s is targetted by %s MCP", nodeName, mcpName)
 			handle, err := profilecreator.NewGHWHandler(mustGatherDirPath, node)
-			cpuInfo, err := handle.CPU()
+			reservedCPUs, isolatedCPUs, err := handle.GetReservedAndIsolatedCPUs(reservedCPUCount, splitReservedCPUsAcrossNUMA)
```

Comment:
It looks like this is business logic and shouldn't be part of the input "validation", so maybe we don't need the "context"...